src/main/java/io/openliberty/tools/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -545,17 +626,41 @@ public void installFeatures(boolean isAcceptLicense, List<String> featuresList)
     	
     	Map<String, String> featureToExtMap = new HashMap<String, String>();
     	List<String> featuresToInstall = new ArrayList<String>();
+    	
+    	DefaultArtifactVersion version;
+    	if(openLibertyVersion != null) {
+    		version = new DefaultArtifactVersion(openLibertyVersion);	
+    		info("plugin listed esa: " + pluginListedEsas.toString());
+    		if (version.compareTo(MIN_VERSION) < 0 && !pluginListedEsas.isEmpty()) {
+    			//manually install user feature esas
+    			info("Neither InstallUtility nor FeatureUtility is available to install user feature esa.");
+    			info("Attempting to manually install the user feature esa without resolving its dependencies.");
+    			info("Recommended user action: upgrade to OpenLiberty version " + MIN_USER_FEATURE_VERSION + " or higher and provide features-bom file for the user feature esa.");
+    			info(" To directly install the esa file without providing features-bom file, upgrade to OpenLiberty version 23.0.0.2.");
+    			
+    			copyUserFeature(pluginListedEsas, installDirectory);
+    		}
+    	}
+    		   	
     	for (String feature: featuresList) {
     		if (feature.contains(":")) {
     			String[] userFeatureSplit = feature.split(":");
-    			featureToExtMap.put(userFeatureSplit[1], userFeatureSplit[0]);
-    			featuresToInstall.add(userFeatureSplit[1]);
+    			String userFeatureName = userFeatureSplit[1].toLowerCase();
+    			featureToExtMap.put(userFeatureName, userFeatureSplit[0]);
+    			//check if user feature esa was installed manually
+    			if(!shortNameMap.containsValue(userFeatureName) && !shortNameMap.containsKey(userFeatureName)) {
```

Comment:
I don't understand why it is checking for the feature name in both the key and value of `shortNameMap`. That map uses the symbolic name for the key. How does this `userFeatureName` map to the `Subsystem-SymbolicName` and the `IBM-ShortName` referenced when populating the `shortNameMap`?